### PR TITLE
provider/aws: Convert AWS RDS to use upstream aws-sdk-go

### PR DESCRIPTION
**Pull request by catsby**
Converts RDS to use github.com/awslabs/aws-sdk-go

### CODE REVIEW

**Comment by catsby** [comment]
rebased on `master`, should be :+1: for review

File: builtin/providers/aws/resource_aws_db_instance.go [hunk 1]
```diff
@@ -5,9 +5,9 @@ import (
 	"log"
 	"time"
 
-	"github.com/hashicorp/aws-sdk-go/aws"
-	"github.com/hashicorp/aws-sdk-go/gen/iam"
-	"github.com/hashicorp/aws-sdk-go/gen/rds"
+	"github.com/awslabs/aws-sdk-go/aws"
+	"github.com/awslabs/aws-sdk-go/service/IAM"
```
Comment (phinze) [inline builtin/providers/aws/resource_aws_db_instance.go]: IAM is really capitalized?

File: builtin/providers/aws/resource_aws_db_parameter_group.go [hunk 2]
```diff
@@ -11,8 +11,8 @@ import (
 	"github.com/hashicorp/terraform/helper/resource"
 	"github.com/hashicorp/terraform/helper/schema"
 
-	"github.com/hashicorp/aws-sdk-go/aws"
-	"github.com/hashicorp/aws-sdk-go/gen/rds"
+        "github.com/awslabs/aws-sdk-go/aws"
+        "github.com/awslabs/aws-sdk-go/service/rds"
```
Comment (phinze) [inline builtin/providers/aws/resource_aws_db_parameter_group.go]: weird whitespace here?

**Comment by phinze** [comment]
Code LGTM - there's a buncha whitespace diff that seems like it's from some weird tab/spaces munging that happened.

**Comment by catsby** [comment]
Fixed the whitespace noise with df45b2c

**Pull request merged**